src/UI/UIView.cpp

```diff
@@ -1348,6 +1806,64 @@ void UIView::addTransferFunction()
   }
   m_tfns.push_back(trfntc);
 }
+
+void UIView::get_Quantiles(int row)
+{
+  m_copy_trnfnct_name = m_tfns[row].Name;
+  float q_min = 0;
+  float q_max = 0;
+  tfn_widget[m_trnfnc_table_selection].get_Quantiles(q_min, q_max);
+  m_histogram_quantiles[0] = q_min;
+  m_histogram_quantiles[1] = q_max;
+  m_trn_fct_options_window = true;
+}
+
+void UIView::set_volume_time_info(time_t time)
+{
+
+  tm* time_info = localtime(&time);
+  if (time_info)
+  {
+    bool pm = time_info->tm_hour >= 12;
+    int hour_12 = (time_info->tm_hour >= 13) ? time_info->tm_hour - 12 : time_info->tm_hour;
+
+    std::stringstream ss_time;
+    ss_time << std::setw(2) << std::setfill('0') << hour_12 << ":";
+    ss_time << std::setw(2) << std::setfill('0') << time_info->tm_min << " ";
+    ss_time << (pm ? "PM" : "AM");
+
+    std::stringstream ss_day;
+    ss_day << months[time_info->tm_mon] << " ";
+    ss_day << std::setw(2) << std::setfill('0') << time_info->tm_mday;
+    ss_day << ", " << time_info->tm_year + 1900;
+
+    m_time_info = ss_time.str();
+    m_day_info = ss_day.str();
+  }
+ 
+
+}
+
+void UIView::draw_tranfer_funciton_legend()
+{
+  if (m_use_transferfunction)
+  {
+    tfn_widget[0].drawLegend();
+  }
+  
+}
+
+void UIView::set_trns_fnct_min_max(float min, float max)
+{
+  /*tfn_widget[0].setMinMax(m_volumes[m_selectedVolume][active_volume]->getMin() * alpha + m_volumes[m_selectedVolume][active_volume2]->getMin() * (1.0 - alpha),
+    m_volumes[m_selectedVolume][active_volume]->getMax() * alpha + m_volumes[m_selectedVolume][active_volume2]->getMax() * (1.0 - alpha));*/
+  if (m_use_transferfunction)
+  {
+    tfn_widget[0].setMinMax(min, max);
+  }
+
+}
+
```

Comment:
this file is very large - is there any way to break it up into smaller componenents?

Reply:
https://github.com/brown-ccv/VR-Volumeviewer/issues/46